server/src/routes.ts

```diff
@@ -2215,6 +2266,480 @@ export default async function routes(fastify: FastifyInstance) {
     }
   );
 
+  //********************************************************************
+  // EVENT TEMPLATES
+  //********************************************************************
+
+  fastify.get(
+    '/event-templates',
+    {
+      schema: {
+        description: 'List paginated event templates, you can filter by name',
+        tags: ['Event Templates',],
+        querystring: {
+          limit: { type: 'number' },
+          offset: { type: 'number' },
+          name: { type: 'string' },
+        },
+      },
+    },
+    async (req: any, res) => {
+      const limit = req.query.limit || 10;
+      const offset = req.query.offset || 0;
+      const name = req.query.name || null;
+
+      let eventTemplates = await getPaginatedEventTemplates(limit, offset, name);
+      const totalEventTemplates = await getTotalEventTemplates(name) || 0;
+
+      return {
+        limit: limit,
+        offset: offset,
+        total: totalEventTemplates,
+        event_template: eventTemplates,
+      };
+    }
+  );
+
+  fastify.get(
+      '/event-templates/:id',
+      {
+        schema: {
+          description: 'Endpoint to get an specific event template',
+          tags: ['Event Templates',],
+          params: {
+            id: { type: 'string' },
+          },
+          response: {
+            200: {
+              type: 'object',
+              properties: {
+                id: { type: 'number' },
+                name: { type: 'string' },
+                title_image: { type: 'string' },
+                title_link: { type: 'string' },
+                header_link_text: { type: 'string' },
+                header_link_url: { type: 'string' },
+                header_color: { type: 'string' },
+                header_link_color: { type: 'string' },
+                main_color: { type: 'string' },
+                footer_color: { type: 'string' },
+                left_image_url: { type: 'string' },
+                left_image_link: { type: 'string' },
+                right_image_url: { type: 'string' },
+                right_image_link: { type: 'string' },
+                mobile_image_url: { type: 'string' },
+                mobile_image_link: { type: 'string' },
+                footer_icon: { type: 'string' },
+                secret_code: { type: 'string' },
+                created_date: { type: 'string' },
+                is_active: { type: 'boolean' },
+              },
+            }
+          },
+          security: [
+            {
+              "authorization": []
+            }
+          ]
```

Comment:
`GET` endpoints should be public. Is this necessary?

Reply:
@nahu90 in both the GET for all (`/`) and for a specific object (`/id`), we should return the secret code for the admins. So maybe the `security` is ok. But I don't se you handling the logic